exporter/redis.go

```diff
@@ -599,6 +629,98 @@ func doRedisCmd(c redis.Conn, cmd string, args ...interface{}) (reply interface{
 	return res, err
 }
 
+var errNotFound = errors.New("key not found")
+
+// getKeyInfo takes a key and returns the type, and the size or length of the value stored at that key.
+func getKeyInfo(c redis.Conn, key string) (info keyInfo, err error) {
+
+	keyType, err := redis.String(c.Do("TYPE", key))
+	if err != nil {
+		return info, err
+	}
+
+	info.keyType = keyType
+
+	switch keyType {
+	case "none":
+		return info, errNotFound
+	case "string":
+		if size, err := redis.Int64(c.Do("PFCOUNT", key)); err == nil {
```

Comment:
I figured there's no need to retrieve the key and check the first few chars before calling `PFCOUNT`; just try it and use it if it doesn't fail.